### PR TITLE
Query legacy-client impl WIP

### PR DESCRIPTION
**Pull request by mfedderly**

### CODE REVIEW

File: packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+    return value.objectSetDefinition;
+  } else if (typeof value === "object") {
+    // Since structs are valid arguments for Queries, we map the values
+    return Object.entries(value).reduce((acc, [key, structValue]) => {
+      acc[key] = getParameterValueMapping(structValue);
+      return acc;
+    }, {} as { [key: string]: PrimitiveParameterValue });
+  }
+
+  return value as string | number | boolean;
```
Comment (mfedderly) [inline packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts:139]: This return type is a bit of a lie, you can get Array<any>, Record<string, any>, and an ObjectSetDefinition

File: packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+): obj is { objectSetDefinition: ObjectSetDefinition } {
+  return obj && obj.objectSetDefinition;
+}
+
+async function remapQueryResponseType(
+  client: ThinClient<OntologyDefinition<any>>,
+  definition: QueryDataTypeDefinition<any>,
+  responseValue: PrimitiveParameterValue,
+): Promise<ParameterValue> {
+  // TODO can the backend really not send us null responses?
```
[the remark this answers is not in the document]
Comment (pashakondratyev) [reply packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts:153]: Yeah that would be good

File: packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+            };
+          });
+
+          return {
+            groups,
+          };
+        }
+
+        case "objectSet":
+          throw new Error("ObjectSet type not supported in response");
```
Comment (pashakondratyev) [inline packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts:312]: We can handle this separately in a FLUP, but we can probably fill this in, just never got around to this in V0

File: packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+      acc[key] = getParameterValueMapping(value);
+      return acc;
+    },
+    parameterMap,
+  );
+
+  return remappedParams;
+}
+
+function getParameterValueMapping(
```
Comment (pashakondratyev) [inline packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts:110]: I think we can share this code with the code in `executeAction`

File: packages/legacy-client/src/client/queries.test.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+          "integer": 0,
+          "long": 0,
+          "boolean": false,
+          "date": nowLocalDate.toISOString(),
+          "string": "",
+          "timestamp": nowTimestamp.toISOString(),
+          "object": {},
+          "objectSet": {
+            "apiName": "Todo",
+            "description": "",
```
Comment (pashakondratyev) [inline packages/legacy-client/src/client/queries.test.ts:106]: I don't think description should be here

File: packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mfedderly) [reply packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts:153]: Lets move it into a flup PR, when I start to mess with the types things seem to be breaking in unexpected ways.

**Review by pashakondratyev — APPROVED** [review]

**Pull request merged**